webknossos-datastore/app/com/scalableminds/webknossos/datastore/storage/TemporaryStore.scala

```diff
@@ -1,36 +1,51 @@
 package com.scalableminds.webknossos.datastore.storage
 
-import akka.actor.ActorSystem
+import akka.actor.{Actor, ActorLogging, ActorSystem}
 import akka.agent.Agent
 import javax.inject.Inject
 
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.stm.Ref
 
 class TemporaryStore[K, V] @Inject()(system: ActorSystem) {
 
-  lazy val ts = Agent[Map[K, V]](Map())
+  lazy val map: scala.collection.mutable.Map[K, V] = scala.collection.mutable.Map()
```

Comment:
Not sure if we should include the changes from #3251 here.

Reply:
We should merge #3251 first, but I think we can do that first thing monday morning :)